src/org/opendatakit/briefcase/util/ExportToCsv.java

```diff
@@ -134,6 +136,46 @@ public boolean doAction() {
 
     File[] instances = instancesDir.listFiles();
 
+		// Sorts the instances by the submission date. If no submission date, we
+		// assume it to be latest.
+		Arrays.sort(instances, new Comparator<File>() {
+			HashMap<File, String> submissionDateToFileMap = new HashMap<File, String>();
+			File submission1, submission2;
+			String submissionDate1String, submissionDate2String;
+			Date submissionDate1, submissionDate2;
+
+			public int compare(File f1, File f2) {
+				try {
+					if (submissionDateToFileMap.get(f1) == null) {
+						submission1 = new File(f1, "submission.xml");
+						submissionDate1String = XmlManipulationUtils.parseXml(submission1).getRootElement()
+								.getAttributeValue(null, "submissionDate");
+						submissionDateToFileMap.put(f1, submissionDate1String);
+					} else {
+						submissionDate1String = submissionDateToFileMap.get(f1);
+					}
+					if (submissionDateToFileMap.get(f2) == null) {

```

Comment:
I would put a blank line before this conditional to indicate that it's a separate section of the code.